cheatsheets/Password_Storage_Cheat_Sheet.md

```diff
@@ -1,567 +1,153 @@
+Additionally, due to how computationally expensive modern hashing functions are, if a user can supply very long passwords then there is a potential denial of service vulnerability, such as the one published in [Django](https://www.djangoproject.com/weblog/2013/sep/15/security/) in 2013.
 
-# Argon2 usage proposal in Java
+In order to protect against both of these issues, a maximum password length should be enforced. This should be 50 characters for Bcrypt, and at least 64 characters for other algorithms. There are two main ways that this can be done:
 
-The objective is to propose a example of secure usage/integration of the Argon2 algorithm in Java application to protect password when stored.
+- Restrict the maximum length in a password policy.
+- Hash the password with another algorithm such as SHA-1 (which produces a 40 character output) or SHA-256 (64 characters) before hashing it with the modern algorithm.
 
-## Argon2 library used
+# Hashing Algorithms
 
-The Argon2 implementation provided by [phc-winner-argon2](https://github.com/P-H-C/phc-winner-argon2) project has been used because:
+## Modern Algorithms
 
-- It's the reference implementation of this algorithm.
-- It's dedicated to this new algorithm so all work by the maintainer is focused on the implementation.
-- Project is active, [last release](https://github.com/P-H-C/phc-winner-argon2/blob/master/CHANGELOG.md) date from december 2017.
-- There bindings for many technologies.
+There are four main algorithms that should be considered for hashing passwords in modern applications. All of these algorithms support an iteration count of [work factor](#work-factors), which should be adjusted based on the system they are being used with.
 
-Java bindings by [phxql](https://github.com/phxql/argon2-jvm) has been used because it's currently the only binding proposed for Java in the [bindings list](https://github.com/P-H-C/phc-winner-argon2#bindings) and is simple to use.
+- [Argon2](https://github.com/P-H-C/phc-winner-argon2) is the winner of the 2015 [Password Hashing Competition](https://password-hashing.net), and should be used as the first choice when it is available. It has strong resistance to both GPU and ASIC based attacks.
+- [Scrypt](https://en.wikipedia.org/wiki/Scrypt) is designed to resist GPU based attacks, but is less widely supported.
+- [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2) is recommended by [NIST](https://pages.nist.gov/800-63-3/sp800-63b.html#memsecretver), and should be used when FIPS compliance is required. It requires that 
```

Comment:
I think this sentence is not finished. Nice mentioning NIST in here!

Reply:
Remove the extra partial sentence.